### PR TITLE
Adds ruff for linting + autoformatting

### LINKED ISSUE #192: Linting with Ruff
Since NOMAD has moved to replacing pycodestyle and pylint with [Ruff](https://docs.astral.sh/ruff/) as the auto-formatter, we should enforce this as well going forward. You can to install Ruff [extensions](https://docs.astral.sh/ruff/integrations/) in your IDE.

### PR DESCRIPTION
**Pull request by domna**
Fixes #192 

Tried to align with nomad. I think it would also make sense to autoformat the code with ruff (but maybe in a separate PR?). Otherwise we will get large diffs as soon as we edit.

### CODE REVIEW

File: pyproject.toml
Comment (domna) [reply pyproject.toml:94]: Correct: https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/blob/develop/pyproject.toml?ref_type=heads#L148

**Comment by sherjeelshabih** [comment]
Before we merge the actual formatted code to this branch, can you add something in the Readme/contribution about `pre-commit install`. Without this done locally for every dev, it won't run the hook right?

**Comment by domna** [comment]
> Before we merge the actual formatted code to this branch, can you add something in the Readme/contribution about `pre-commit install`. Without this done locally for every dev, it won't run the hook right?

Correct. I'll add it

**Review by lukaspie — APPROVED** [review]
LGTM (after #201 is merged).

**Pull request merged**
